### ISSUE TITLE
Support for sybase data types

### ISSUE THREAD

**Issue by nrhy** [blue-yonder/turbodbc#161]
Hello,

I'm trying to adapt turbodbc to sybase's (SAP) Adaptive Server Enterprise driver - while I have no problem using pyodbc, I really wanted to leverage pyarrow/numpy/asyncio functionalities that turbodbc offers.

It appears that for most data types everything comes out alright, however, I've found cases where Binary types are not well supported.

Errors found so far:

RuntimeError: Error! Unsupported type identifier for column timestamp @ NULLABLE SQL_VARBINARY (precision 8, scale 0))

Error! Unsupported type identifier for column claim_id @ SQL_BINARY (precision 16, scale 0))

Driver:

Adaptive Server Enterprise

p.s I really love this library, so thanks a lot for putting it together! On the flip side however, what are the chances of turbodbc joining efforts with the pyodbc team? (had to ask)

**Comment by MathMagique** [comment]
Hi! Thanks for reporting your findings! I will, however, close this issue as a duplicate of #56. The binary type is pretty much up the list of features I would like to add next, but time is an issue right now.

Concerning your meta-question, chances of the pyodbc team (essentially Michael Kleehammer) and the turbodbc team (essentially myself) joining forces are rather slim. The code bases are very different, here are just a few differences:

* Modern C++ vs. C-style programming
* Pybind11 vs. Python C-API usage
* Batched data transfer vs. single-row statements
* Well-developed build infrastructure vs. ad-hoc development/testing
* Modern columnar API vs. standard row-based API

That's not saying that pyodbc is bad or anything; it is battle-tested, highly compatible and is used by many production systems. Turbodbc still has to play catching up in a few regards. It is, however, not easy to port over turbodbc achievements into the pyodbc codebase or vice versa.

**Comment by MathMagique** [comment]
@nrhy The Python C-API is relatively far away from object-orientation, let alone C++11. It also requires developers to handle reference counting, which is a real pain to get right.

Pybind11 allowed me to expose my C++11 code base to Python with little hassle and with the same code for Python 2 and 3. Previously I used Boost Python, which meant more hassle and did not easily allow me to test my code with both Python 2 and 3 on my Mac. I could have used other ways to write the driver, e.g., with cffi, ctypes, cython, but in the end I chose the tool that allowed me to write C++11 code with the least issues. Packaging sucked, though ;-).
